### DISCUSSION TITLE
algebraic and initial conditions must have the same shape after discretisation but algebraic.shape = (20, 1) and initial_conditions.shape = (21, 1) for variable 'Catalyst layer Temperature of anode [K]'.

### DISCUSSION THREAD

**Discussion by lwang95** [pybamm-team/PyBaMM#3626] [Q&A]
I want to slove the problem with differnet domains. But I cannot make the discretisation in the solver. the shape of algebraic is always bigger than the shape of the initial_conditions. How can I solve the problem?
![image](https://github.com/pybamm-team/PyBaMM/assets/81779120/b7a2e20b-a060-48bf-8c49-abbc058d7f07)
![image](https://github.com/pybamm-team/PyBaMM/assets/81779120/4b0fe409-32e2-4881-839c-f4f9dab6a967)
![image](https://github.com/pybamm-team/PyBaMM/assets/81779120/03638d7d-9115-4919-90e6-ba990da7c1a9)
![image](https://github.com/pybamm-team/PyBaMM/assets/81779120/a8ba3294-e1a1-4f3e-8e7e-beb452b3e998)
![image](https://github.com/pybamm-team/PyBaMM/assets/81779120/bf9ee70c-aac4-4303-b7e2-97185fb43d9d)

**Comment by rtimms** [comment]
This is because in the Finite Volume Method variables are evaluated on cell centres and gradients (fluxes) on cell edges. So the `grad` terms are of a different shape (off-by-one) compared to other terms. Can you share the source of the equations you are using in the model? It seems you are mixing scalar and vector quantities in the same equation (`grad` is really a vector).

**Reply by lwang95** [reply]
So, I think those are all of my equations. The equations now are not exactly correct. I want to make the 1-D stationary thermal model of the cell. Now I just try how to solve the equtions in different domains.

**Reply by lwang95** [reply]
![image](https://github.com/pybamm-team/PyBaMM/assets/81779120/6f07029d-ba67-4501-a63f-4669cf7e9a44)
![image](https://github.com/pybamm-team/PyBaMM/assets/81779120/000bf38f-5d7c-4d3f-b977-51eb7a512621)
![image](https://github.com/pybamm-team/PyBaMM/assets/81779120/5c5b5a21-6367-4731-af79-7a2ac6cfece1)
